code/modules/reagents/newchem/medicine.dm

```diff
@@ -0,0 +1,711 @@
+	color = "#C8A5DC"
+	metabolization_rate = 0.2
+	overdose_threshold = 30
+
+datum/reagent/epinephrine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.health < -10 && M.health > -65)
+		M.adjustToxLoss(-1*REM)
+		M.adjustBruteLoss(-1*REM)
+		M.adjustFireLoss(-1*REM)
+	if(M.oxyloss > 35)
+		M.setOxyLoss(35)
+	if(M.losebreath >= 4)
+		M.losebreath -= 4
+	if(M.losebreath < 0)
+		M.losebreath = 0
+	if(prob(30))
+		M.AdjustParalysis(-1)
+		M.AdjustStunned(-1)
+		M.AdjustWeakened(-1)
+	..()
+	return
+
+datum/reagent/epinephrine/overdose_process(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(3*REM)
+		M.losebreath++
+	..()
+	return
+
+/datum/chemical_reaction/epinephrine
+	name = "Epinephrine"
+	id = "epinephrine"
+	result = "epinephrine"
+	required_reagents = list("phenol" = 1, "acetone" = 1, "diethylamine" = 1, "oxygen" = 1, "chlorine" = 1, "hydrogen" = 1)
+	result_amount = 6
+
+/datum/reagent/mannitol/on_mob_life(mob/living/M as mob)
+	M.adjustBrainLoss(-3)
+	..()
+	return
+
+/datum/chemical_reaction/mannitol
+	name = "Mannitol"
+	id = "mannitol"
+	result = "mannitol"
+	required_reagents = list("sugar" = 1, "hydrogen" = 1, "water" = 1)
+	result_amount = 3
+	mix_message = "The solution slightly bubbles, becoming thicker."
+
+/datum/reagent/mannitol
+	name = "Mannitol"
+	id = "mannitol"
+	description = "Heals brain damage effectively. Use it in cyro tubes alongside Cryoxadone."
+	color = "#C8A5DC"
+
+/datum/reagent/mutadone/on_mob_life(var/mob/living/carbon/human/M as mob)
+
+	if(!holder) return
+	if(!M) M = holder.my_atom
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.hulk_time = 0
+	var/needs_update = M.mutations.len > 0
+	for(var/datum/dna/gene/G in dna_genes)
+		if(G.is_active(M))
+			if(G.name == "Hulk" && ishuman(M))
+				G.OnMobLife(M)
+			var/tempflag = 0
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				if(H.species && (G.block in H.species.default_blocks))
+					tempflag |= GENE_NATURAL
+			if(G.can_deactivate(M, tempflag))
+				G.deactivate(M,0, tempflag)
+	M.alpha = 255
+	//M.mutations = list()
+	//M.active_genes = list()
+
+	M.disabilities = 0
+	M.sdisabilities = 0
+
+	//Makes it more obvious that it worked.
+	M.jitteriness = 0
+
+	// Might need to update appearance for hulk etc.
+	if(needs_update && ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.update_mutations()
+	..()
+	return
+
+/datum/chemical_reaction/mutadone
+	name = "Mutadone"
+	id = "mutadone"
+	result = "mutadone"
+	required_reagents = list("mutagen" = 1, "acetone" = 1, "bromine" = 1)
+	result_amount = 3
+
+
+/datum/reagent/mutadone
+	name = "Mutadone"
+	id = "mutadone"
+	description = "Heals your genetic defects."
+	color = "#C8A5DC"
+
+datum/reagent/antihol
+	name = "Antihol"
+	id = "antihol"
+	description = "Helps remove Alcohol from someone's body, as well as eliminating its side effects."
+	color = "#C8A5DC"
+
+datum/reagent/antihol/on_mob_life(var/mob/living/M as mob)
+	M.dizziness = 0
```

Comment:
no real way to tell if it was caused by alcohol other than by reducing it only if you have booze in the system, which doesn't accomplish anything because it'll just take 1 step forward 1 step back